sunpy/map/mapbase.py

```diff
@@ -1572,6 +1579,31 @@ def draw_rectangle(self, bottom_left, width: u.deg, height: u.deg, axes=None, **
         else:
             axes_unit = self.spatial_units[0]
 
+        if width or height:
+            warnings.warn("Use of `width` and `height` is depreciated and these will be removed in future", Warning)
```

Comment:
The warning here I think should be a sunpy deprecated warning

Reply:
> Minor comments. The issue seems to be that
> when the tests get to this (line 1612)
> 
> ```
>         width = width.to(axes_unit).value
>         height = height.to(axes_unit).value
> ```
> 
> Height is set to none and is not allowed to be none. Somewhere you need to set this.

So I found the reason. This was because of the ordering of the parameters. If I put `top_right` at the last, this issue is resolved. As it is, because the tests do not name the parameters while passing them, height defaults to `None`, besides there is a mismatch of the units in `top_right` and `bottom_left`.
However, I believe we should change the test as putting two deprecated parameters in between two "in use" parameters in the function parameter list would not make sense.

Reply:
For some reason using `SunpyPendingDeprecationWarning`
gives 
`name 'SunpyPendingDeprecationWarning' is not defined`
even though the class is defined in `sunpy.util.exceptions.py`

Reply:
Yeah you will need to import it at the top of the file.

Reply:
Whoops. My bad. :sweat_smile: